src/util.hpp

```diff
@@ -29,5 +29,42 @@ inline char* to_csv(const std::vector<std::string>& names) {
   return strdup(s_c);
 }
 
+/**
+ * Convert exception-style error handling into our C output parameter
+ * style. F is always a lambda, so this code gets inlined and the function
+ * call overhead is optimized away.
+ */
+template <typename F>
+[[gnu::always_inline]] [[msvc::forceinline]]
+inline auto handle_errors(const char* name, char** error_msg, F f) {
+  try {
+    return f();
+  } catch (const std::exception& e) {
+    if (error_msg) {
```

Comment:
Don't we want to print `e.what()` even if `error_msg` is null?  And provide a message no matter what, even if there's no `error_msg` or `e.what()`?

Reply:
That hasn’t been our behavior so far. If the user wanted to see error messages, they shouldn’t have provided us a null pointer for the error message argument